### PR TITLE
feat(multiframe): enhanced support for multiframe dicom

### PR DESCRIPTION
**Pull request by rodrigobasilio2022**
Changes made to OHIF so I can open multiframe series

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: extensions/cornerstone/package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "peerDependencies": {
     "@ohif/core": "^3.0.0",
     "@ohif/ui": "^2.0.0",
-    "cornerstone-wado-image-loader": "^4.2.1",
+    "cornerstone-wado-image-loader": ">=4.8.0",
```
Comment (wayfarer3130) [inline extensions/cornerstone/package.json:32]: This isn't the right import any more - should be @cornerstonejs/...wado image loader now.

File: extensions/default/src/DicomWebDataSource/index.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ function createDicomWebApi(dicomWebConfig, UserAuthenticationService) {
     query: {
       studies: {
         mapParams: mapParams.bind(),
-        search: async function (origParams) {
```
Comment (wayfarer3130) [inline extensions/default/src/DicomWebDataSource/index.js:107]: If you aren't changing a file, please don't reformat it.

File: platform/core/package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
   },
   "peerDependencies": {
     "cornerstone-math": "0.1.9",
-    "cornerstone-wado-image-loader": "^4.2.1",
+    "cornerstone-wado-image-loader": ">=4.8.0",
```
Comment (wayfarer3130) [inline platform/core/package.json:34]: Ditto

File: platform/core/src/services/DisplaySetService/extractFrameMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Extract frame metadata from a multiframe metadata. All information related to
+ * multiframe is removed from the instance metadata
+ */
+const extractFrameMetadata = (frame, instance) => {
```
Comment (wayfarer3130) [inline platform/core/src/services/DisplaySetService/extractFrameMetadata.ts:5]: I thought this code already existed somewhere?  If it does, we should have a single copy.  If you are just moving it, please document that - at least in a comment.

File: extensions/default/src/DicomLocalDataSource/index.js [hunk 5]
```diff
@@ -153,6 +155,7 @@ function createDicomLocalApi(dicomLocalConfig) {
                 StudyInstanceUID,
                 SeriesInstanceUID,
                 SOPInstanceUID,
+                frameIndex: isMultiframe ? index : undefined,
```
Comment (wayfarer3130) [inline extensions/default/src/DicomLocalDataSource/index.js:158]: Is there any reason to not include frameIndex if this is an image object?  An index of 1 is the correct index for images all the time, even for non-multiframes.

File: platform/core/src/classes/MetadataProvider.js [hunk 6]
```diff
@@ -445,15 +445,16 @@ class MetadataProvider {
...
     if (urlRegex.test(imageId)) {
       imageURI = imageId;
     } else {
       imageURI = imageIdToURI(imageId);
     }
 
     const uids = this.imageURIToUIDs.get(imageURI);
-    const frameNumber = imageId.split(/\/frames\//)[1];
+    const frameNumber =
+      imageId.split(/\/frames\//)[1] || imageId.split('&frame=')[1];
```
Comment (wayfarer3130) [inline platform/core/src/classes/MetadataProvider.js:457]: This doesn't work as you expect unless the frames/{...} or the &frame= is the last thing in the URL, but there is no guarantee that is so - particularly for &frame=, that could be anywhere in the URL.  You should write a URL that matches a sub-part, and then use that to split things up.

File: platform/viewer/public/config/default.js [hunk 7]
```diff
@@ -30,10 +30,15 @@ window.config = {
...
         // new server
-        wadoUriRoot: 'https://domvja9iplmyu.cloudfront.net/dicomweb',
-        qidoRoot: 'https://domvja9iplmyu.cloudfront.net/dicomweb',
-        wadoRoot: 'https://domvja9iplmyu.cloudfront.net/dicomweb',
+        //wadoUriRoot: 'https://domvja9iplmyu.cloudfront.net/dicomweb',
+        //qidoRoot: 'https://domvja9iplmyu.cloudfront.net/dicomweb',
+        //wadoRoot: 'https://domvja9iplmyu.cloudfront.net/dicomweb',
+        wadoUriRoot: 'http://localhost/dicom-web',
+        qidoRoot: 'http://localhost/dicom-web',
+        wadoRoot: 'http://localhost/dicom-web',
```
Comment (wayfarer3130) [inline platform/viewer/public/config/default.js:40]: wadoUriRoot is redundant at this point unless it is actually using wadoUri

File: extensions/cornerstone/package.json (hunk 1 above)
Comment (rodrigobasilio2022) [reply extensions/cornerstone/package.json:32]: I just changed the version? Are you confusing with  "@cornerstonejs/streaming-image-volume-loader"?

File: platform/core/package.json (hunk 3 above)
Comment (rodrigobasilio2022) [reply platform/core/package.json:34]: I just changed the version. Are you confusing with "@cornerstonejs/streaming-image-volume-loader"?

File: platform/core/src/services/DisplaySetService/extractFrameMetadata.ts (hunk 4 above)
Comment (rodrigobasilio2022) [reply platform/core/src/services/DisplaySetService/extractFrameMetadata.ts:5]: This was a temporary file erased in this commit

File: platform/viewer/public/config/default.js (hunk 7 above)
Comment (rodrigobasilio2022) [reply platform/viewer/public/config/default.js:40]: This was a temporary config for local dicomweb server. Changed in the next commit

File: extensions/default/src/DicomLocalDataSource/index.js (hunk 5 above)
Comment (rodrigobasilio2022) [reply extensions/default/src/DicomLocalDataSource/index.js:158]: done

**Review by sedghi — APPROVED** [review]

**Pull request merged**